pgjdbc/src/main/java/org/postgresql/core/QueryExecutorBase.java

```diff
@@ -52,6 +55,10 @@
   private final LruCache<Object, CachedQuery> statementCache;
   private final CachedQueryCreateAction cachedQueryCreateAction;
 
+  // For getParameterStatuses(), GUC_REPORT tracking
+  private final TreeMap<String,String> parameterStatuses
+      = new TreeMap<String,String>(String.CASE_INSENSITIVE_ORDER);
```

Comment:
Why is it a `TreeMap`?
Should the field be just `Map` and the value just `HashMap`?

Reply:
`TreeMap` is used to permit `String.CASE_INSENSITIVE_ORDER`, which is not supported by `HashMap`. That ensures that parameter lookups will find the parameter no matter what case is used - `datestyle` or `DateStyle` for example.

Reply:
Pg's parameters use a mix of camel case and underscore separators. Users generally expect case-folding behaviour in postgres, too. So I'd strongly prefer to do this.